web/docs/project/env-vars.md

```diff
@@ -217,21 +218,156 @@ We talk about how to define env vars for each deployment option in the [deployme
 
 ## Custom Env Var Validations
 
-TODO: when the Zod validation PRs are merged, describe how users can define their own validations.
+If your code requires some environment variables, you usually want to ensure that they are correctly defined. In Wasp, you can define your environment variables validation by defining a [Zod object schema](https://zod.dev/?id=basic-usage) and telling Wasp to use it.
+
+:::info What is Zod?
+
+[Zod](https://zod.dev/) is a library that lets you define what you expect from your data. For example, you can use Zod to define that a value should be a string that's a valid email address or that a value should be a number between 0 and 100.
+
+:::
+
+Let's look at an example of defining env vars validation:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/env.js"
+import * as z from 'zod'
+
+export const serverEnvValidationSchema = z.object({
+  STRIPE_API_KEY: z.string({
+    required_error: 'STRIPE_API_KEY is required.',
+  }),
+})
+
+export const clientEnvValidationSchema = z.object({
+  REACT_APP_NAME: z.string().default('TODO App'),
+})
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/env.ts"
+import * as z from 'zod'
+
+import { defineEnvValidationSchema } from 'wasp/env'
+
+export const serverEnvValidationSchema = defineEnvValidationSchema(
+  z.object({
+    STRIPE_API_KEY: z.string({
+      required_error: 'STRIPE_API_KEY is required.',
+    }),
+  })
+)
+
+export const clientEnvValidationSchema = defineEnvValidationSchema(
+  z.object({
+    REACT_APP_NAME: z.string().default('TODO App'),
+  })
+)
+```
+
+We are using the `defineEnvValidationSchema` helper to get type-checking for the Zod schema.
```

Comment:
I suggest changing the subject to emphasize a point and address the user directly:
```suggestion
The defineEnvValidationSchema` function ensures your Zod schema is type-checked.
```